tools/ideabuck/src/com/facebook/buck/intellij/ideabuck/run/BuckRunConfigurationExtension.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+public class BuckRunConfigurationExtension extends RunConfigurationExtension {
+
+  private String targetResult;
+  private List<String> additionalClasspath;
+  private Project project;
+
+  @Override
+  public <T extends RunConfigurationBase> void updateJavaParameters(T configuration,
+      JavaParameters javaParameters, RunnerSettings runnerSettings) throws ExecutionException {
+    Project currentProject = configuration.getProject();
+    VirtualFile containingFile;
+    VirtualFile buckFile;
+    if (configuration instanceof ApplicationConfiguration) {
+      containingFile = ((ApplicationConfiguration) configuration).getMainClass()
+          .getContainingFile().getVirtualFile();
+      buckFile = BuckFileUtil.getBuckFile(containingFile);
+    } else {
+      return;
+    }
+
+    BuckBuildManager buildManager = BuckBuildManager.getInstance(currentProject);
+    BuckQueryCommandHandler buckQueryCommandHandler =
+        new BuckQueryCommandHandler(
+            currentProject,
+            buckFile.getParent(),
+            BuckCommand.QUERY,
+            new Function<List<String>, Void>() {
+              @javax.annotation.Nullable
+              @Override
+              public Void apply(@javax.annotation.Nullable List<String> strings) {
+                if (strings == null
+                    || strings.isEmpty()
+                    || strings.get(0) == null
+                    || strings.get(0).isEmpty()) {
+                  return null;
+                }
+                String targetResultCandidate = strings.get(0);
+                for (String candidate : strings) {
+                  if (!candidate.contains("lint")) {
+                    targetResultCandidate = candidate;
+                  }
+                }
+                targetResult = targetResultCandidate;
+                return null;
+              }
+            });
+    buckQueryCommandHandler.command().addParameter("owner(" + containingFile.getPath() + ")");
+    buildManager.runInCurrentThreadPostEnd(
+        buckQueryCommandHandler,
+        () -> {
+          if (targetResult != null) {
+            BuckModule buckModule = currentProject.getComponent(BuckModule.class);
+            buckModule.attach(targetResult);
+
+            ResultCallbackBuckHandler buckAuditCommandHandler =
+                new ResultCallbackBuckHandler(
+                    currentProject,
+                    buckFile.getParent(),
+                    BuckCommand.AUDIT,
+                    new FutureCallback<String>() {
+                      @Override
+                      public void onSuccess(@Nullable String result) {
+                        if (result == null
+                            || result.isEmpty()) {
+                          return;
+                        }
+                        String lines[] = result.split("\\r?\\n");
```

Comment:
`buck query` can output in json format, can you use it instead of parsing new lines?